src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -585,6 +610,65 @@ def alembic(self) -> Alembic:
             raise ValueError("Store not initialized")
         return self._alembic
 
+    @classmethod
+    def filter_and_paginate(
+        cls,
+        session: Session,
+        query: Union[Select[AnySchema], SelectOfScalar[AnySchema]],
+        table: Type[AnySchema],
+        filter_model: FilterBaseModel,
+    ) -> Page[B]:
+        """Given a query, return aPage instance with a list of filtered Models.
+
+        Args:
+            session: The SQLModel Session
+            query: The query to execute
+            table: The table to select from
+            filter_model: The filter to use, including pagination and sorting
+
+        Returns:
+            The Domain Model representation of the DB resource
+        """
+
+        # Filtering
+        filters = filter_model.generate_filter(table=table)
+
+        if filters is not None:
+            query = query.where(filters)
+
+        # Sorting
+        query = query.order_by(getattr(table, filter_model.sort_by))
+
+        # Get the total amount of items in the database for a given query
+        total = session.scalar(
+            select(func.count("*")).select_from(
+                query.order_by(None).options(noload("*")).subquery()
+            )
+        )
+
+        # Get the total amount of pages in the database for a given query
+        total_pages = math.ceil(total / filter_model.size)
+
+        # Get a page of the actual data
+        item_schemas: List[AnySchema] = (
+            session.exec(
+                query.limit(filter_model.size).offset(filter_model.offset)
+            )
+            .unique()
```

Comment:
I suspect this hidden `unique` call might create some hard to debug issues in the future, what is this needed for actually?

Reply:
@AlexejPenner